python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -1149,19 +1155,15 @@ def test_fixed_size_bytes_does_not_accept_varying_lengths(self):
 
     def test_variable_size_bytes(self):
         s = pd.Series([b'123', b'', b'a', None])
-        arr = pa.Array.from_pandas(s, type=pa.binary())
-        assert arr.type == pa.binary()
         _check_series_roundtrip(s, type_=pa.binary())
 
     def test_binary_from_bytearray(self):
-        s = pd.Series([bytearray(b'123'), bytearray(b''), bytearray(b'a')])
+        s = pd.Series([bytearray(b'123'), bytearray(b''), bytearray(b'a'),
+                       None])
         # Explicitly set type
-        arr = pa.Array.from_pandas(s, type=pa.binary())
-        assert arr.type == pa.binary()
-        # Infer type from bytearrays
-        arr = pa.Array.from_pandas(s)
-        assert arr.type == pa.binary()
         _check_series_roundtrip(s, type_=pa.binary())
+        # Infer type from bytearrays
+        _check_series_roundtrip(s)
```

Comment:
But you should pass `expected_pa_type` here.

Reply:
ooops, right - thanks for catching that!